File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.cpp
class Solution {
public:
    vector<int> rearrangeArray(vector<int>& nums) {
        vector<int>pos;
        vector<int> neg;
        int i=0,j=0;

        for(int i=0;i<nums.size();i++){
            if(nums[i]>0)
            pos.push_back(nums[i]);
            else
            neg.push_back(nums[i]);

        }
    
     while(j<nums.size()){
         nums[j]=pos[i];
         j++;
         nums[j]=neg[i];
         i++;
         j++;
     }
     return nums;
    }
};

File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.java
class Solution {
    public int[] rearrangeArray(int[] nums) {
        int n=nums.length;

        int[] ans=new int[n];

        int negative=1,index=0,positive=0;

        while(index<n){
            if(nums[index]>0){
                ans[positive]=nums[index];
                positive+=2;
            }
            else{
                ans[negative]=nums[index];
                negative+=2;
            }
            index++;
        }

        return ans;

    }
}